### PR TITLE
Refactor protocol to objects

### PR DESCRIPTION
**Pull request by hpihkala**
Refactoring `streamr-client` to use the new protocol classes from shared `streamr-client-protocol` package. More code removed than added.

### CODE REVIEW

**Comment by hpihkala** [comment]
@Aapzu do you have time to review this?

File: src/Connection.js [hunk 1]
```diff
@@ -56,13 +57,16 @@ class Connection extends EventEmitter {
...
-                const decodedWrapper = decodeBrowserWrapper(messageEvent.data)
-                const decodedMessage = decodeMessage(decodedWrapper.type, decodedWrapper.msg)
-                this.emit(decodedWrapper.type, decodedMessage, decodedWrapper.subId)
+                websocketResponse = WebsocketResponse.deserialize(messageEvent.data)
             } catch (err) {
                 this.emit('error', err)
             }
+
+            if (websocketResponse) {
+                this.emit(websocketResponse.constructor.getMessageName(), websocketResponse)
```
Comment (Aapzu) [inline src/Connection.js]: If this was moved to the `try` block the `if` wouldn't be needed

File: src/StreamrClient.js [hunk 2]
```diff
@@ -39,54 +45,50 @@ export default class StreamrClient extends EventEmitter {
...
-                        .forEach((sub) => {
-                            sub.setState(Subscription.State.subscribed)
-                        })
-                }
+        this.connection.on('SubscribeResponse', (response) => {
+            const subs = this.subsByStream[response.payload.streamId]
 
-                debug('Client subscribed: %o', response)
+            // The typeof array === 'object'
+            if (subs && typeof subs === 'object') {
```
Comment (Aapzu) [inline src/StreamrClient.js]: If you want to check whether `subs` is an array or not please use `Array.isArray` instead:)
```suggestion
            if (Array.isArray(subs)) {
```
No need for additional nullcheck that way (`Array.isArray(undefined) === false`)

File: package.json [hunk 3]
```diff
@@ -63,6 +63,7 @@
     "eventemitter3": "^3.0.1",
     "node-fetch": "^2.1.2",
     "querystring": "^0.2.0",
+    "streamr-client-protocol": "^1.1.0",
```
[the remark this answers is not in the document]
Comment (hpihkala) [reply package.json]: Done

**Pull request merged**